src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/HearingDocumentService.java

```diff
@@ -132,4 +147,82 @@ public void sendToBulkPrint(CaseDetails caseDetails, String authorisationToken)
 
         return caseDocuments;
     }
+
+    public void createAndSendAdditionalHearingDocuments(String authorisationToken, CaseDetails caseDetails) throws IOException {
+        CaseDocument document = generateAdditionalHearingDocument(caseDetails, authorisationToken);
+        addAdditionalHearingDocumentToCaseData(caseDetails, document);
+        bulkPrintAdditionalHearingDocuments(caseDetails, authorisationToken);
+    }
+
+    private CaseDocument generateAdditionalHearingDocument(CaseDetails caseDetails, String authorisationToken) throws IOException {
+        log.info("Generating Additional Hearing Document for Case ID: {}", caseDetails.getId());
+        CaseDetails caseDetailsCopy = documentHelper.deepCopy(caseDetails, CaseDetails.class);
+        prepareCaseDetailsForDocumentGeneration(caseDetailsCopy);
+
+        return genericDocumentService.generateDocument(authorisationToken, caseDetailsCopy,
+            documentConfiguration.getAdditionalHearingTemplate(), documentConfiguration.getAdditionalHearingFileName());
+    }
+
+    private void prepareCaseDetailsForDocumentGeneration(CaseDetails caseDetails) throws IOException {
+        Map<String, Object> caseData = caseDetails.getData();
+        Map<String, Object> courtDetailsMap = objectMapper.readValue(getCourtDetailsString(), HashMap.class);
+        Map<String, Object> courtDetails = (Map<String, Object>)
+            courtDetailsMap.get(caseData.get(CaseHearingFunctions.getSelectedCourt(caseData)));
+
+        FrcCourtDetails selectedFRCDetails = FrcCourtDetails.builder()
+            .courtName((String) courtDetails.get(COURT_DETAILS_NAME_KEY))
+            .courtAddress((String) courtDetails.get(COURT_DETAILS_ADDRESS_KEY))
+            .phoneNumber((String) courtDetails.get(COURT_DETAILS_PHONE_KEY))
+            .email((String) courtDetails.get(COURT_DETAILS_EMAIL_KEY))
+            .build();
+
+        caseData.put("ccdCaseNumber", caseDetails.getId());
+        caseData.put("DivorceCaseNumber", caseDetails.getData().get(DIVORCE_CASE_NUMBER));
+        caseData.put("ApplicantName", buildFullApplicantName(caseDetails));
```

Comment:
we would get a lot of these values if we called the prepareLetterTemplateData method in DocumentHelper. I'm keen that all of our letters call this reusable method to add generic values like names, address details, court details and case numbers. then only add values that are specific to this document in here.

Reply:
Initially I had used this, however as the dcoument is actually not a letter I decided against using these methods and having them in my own prep method